frontend/src/components/header/MenuOverlay.js

```diff
@@ -91,10 +91,13 @@ class MenuOverlay extends Component {
   handleQuery = (e) => {
     e.preventDefault();
 
-    if (e.target.value) {
+    const targetValue = leftTrim(e.target.value);
```

Comment:
why leftTrim and not simply trim?

Reply:
because we want to trim spaces from the beginning of the string. We want to allow strings to be aded after a string that has space. 
i.e: you type 'Sales '  (sales and space) and then Order
makes sense ?